src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/UdtTest/UdtTest2.cs

```diff
@@ -588,6 +588,128 @@ public void TestSqlUserDefinedAggregateAttributeMaxByteSize()
                 () => create(-2),
                 errorMessage);
         }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsUdtTestDatabasePresent), nameof(DataTestUtility.AreConnStringsSetup))]
+        public void UDTParams_DeriveParameters_CheckAutoFixSuccess()
+        {
+            // the type and sproc must be commited to the database or this test will deadlock with a schema lock violation
+            // if you are missing these database entities then you should look for an updated version of the database creation script
+
+            string sprocName = "sp_insert_customers";
+            string typeName = "CustomerAddress";
+            string customerAddressTypeIncorrectName = $"{DataTestUtility.UdtTestDbName}.dbo.{typeName.Trim('[', ']')}";
+            string customerAddressTypeCorrectedName = $"[dbo].[{typeName.Trim('[', ']')}]";
+            string customerParameterName = "@customers";
+
+            Address addr = Address.Parse("123 baker st || Redmond");
+            DataTable table = new DataTable();
+            table.Columns.Add();
+            table.Columns.Add();
+            table.Rows.Add("john", addr);
+
+            using (SqlConnection connection = new SqlConnection(_connStr))
+            {
+                connection.Open();
+                using (SqlTransaction transaction = connection.BeginTransaction())
+                using (SqlCommand cmd = new SqlCommand(sprocName, connection, transaction))
+                {
+                    try
+                    {
+                        cmd.CommandType = CommandType.StoredProcedure;
+
+                        SqlCommandBuilder.DeriveParameters(cmd);
+
+                        Assert.NotNull(cmd.Parameters);
+                        Assert.Equal(2, cmd.Parameters.Count); // [return_value, table]
+
+                        SqlParameter p = cmd.Parameters[1];
+
+                        Assert.Equal(customerParameterName, p.ParameterName);
+                        Assert.Equal(SqlDbType.Structured, p.SqlDbType);
+                        Assert.Equal(customerAddressTypeIncorrectName, p.TypeName); // the 3 part name is incorrect but needs to be maintained for compatibility
+                        p.Value = table;
+
+                        cmd.ExecuteNonQuery();
+
+                        Assert.Equal(customerAddressTypeCorrectedName, p.TypeName); // check that the auto fix has been applied correctly
+                    }
+                    finally
+                    {
+                        try
+                        {
+                            transaction.Rollback();
+                        }
+                        catch
+                        {
+                            // ignore rollback failure exceptions to preserve original thrown error in test result
```

Comment:
I'm not in favor of "unexpected" test behaviors, that just makes it unreliable code.
If this is an "Expected" exception can we assert the condition it would occur on?
Otherwise if it should NOT occur, we need to fail here.

Secondly, when you say "_to preserve original thrown error in test result_" can you elaborate which exception you're talking about here? I don't see any other exception being throws/asserted.

Reply:
We want the error from the com.ExecuteNonQuery to be the result of executing the command but if the transaction throws an exception because there's been an error rolling back because nothing was done then I want to ignore it, otherwise an exception thrown in the finally is the result instead of the exception from the try, it's confusing.

I'm not in favour of unreliable either but it's defensive. If something were expected it would be dealt with specifically. I'm pretty sure the test body was copied from another test and altered to fit this specific case, it was 2 years ago though do I can't be certain.

Reply:
But it should not throw exception (in ideal scenarios), so it should be removed. It's probably bad old practice to swallow unexpected behaviors..

I was also trying to run this test class manually and was facing some issues, did u manage to run this manually too?
Currently test lab does not run these tests, so relying on checks wouldn't be a good idea for UDT tests. That's one major limitation in our test suite we have in backlog.

Reply:
You'll have to drop and re-create your udt test database using the updated script from this PR. otherwise you won't have the required entities in the database, as the comment at the top of the test says.

Yes, I ran both tests under commandline build and in visual studio successfully. Doing so the first time showed I'd made a mistake that needed correcting so I know they were run repeatedly under successful.

Reply:
Ok. I didn't notice the SQL file changes, will try that. Thanks! :)